### PR TITLE
2587 certificates unitofwork

### PR DESCRIPTION
**Pull request by tnickelsen**

### CODE REVIEW

**Review by TopSwagCode — APPROVED** [review]

File: domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs [hunk 1]
```diff
@@ -70,7 +74,20 @@ public async Task<CreateContractResult> Create(string gsrn, string meteringPoint
 
         try
         {
-            await repository.Save(contract);
+            await unitOfWork.CertificateIssuingContractRepo.Save(contract);
+            await unitOfWork.ActivityLogEntryRepo.AddActivityLogEntryAsync(ActivityLogEntry.Create(
```
Comment (TopSwagCode) [inline domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs:78]: await unitOfWork.ActivityLogEntryRepo.AddActivityLogEntryAsync(ActivityLogEntry.Create(

Shouldn't this be part of UOW aswell???

Comment (TopSwagCode) [reply domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs:78]: We would need transaction scope to ensure that ActivityLog is only saved if whole transaction goes fine.

Comment (tnickelsen) [reply domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs:78]: It is as I am using the repo from the unit of work?

Why? SaveChanges is called to ensure this

Comment (TopSwagCode) [reply domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs:78]: await unitOfWork.ActivityLogEntryRepo.AddActivityLogEntryAsync(ActivityLogEntry.Create()) <-- just don't understand why this is Async then?

**Review by TopSwagCode — APPROVED** [review]

**Pull request merged**